### ISSUE TITLE
Cluster upgrade issue

### ISSUE THREAD

**Issue by dzadonskii** [vitobotta/hetzner-k3s#146]
When executing upgrade command with K3S minor version higher than 9 I get "New k3s version must be more recent than current version" error.

I guess it can be related to how versions array is organized and compared (https://github.com/vitobotta/hetzner-k3s/blob/decb3f25a51d5e9af5e40013b80038af14c9105b/src/configuration/settings/new_k3s_version.cr#L32):

```
v1.21.13-rc1+k3s1
v1.21.14+k3s1
v1.21.14-rc1+k3s1
v1.21.2+k3s1
v1.21.2-rc1+k3s1
v1.21.2-rc1+k3s2
v1.21.2-rc2+k3s2
v1.21.3+k3s1
```

1.21.14 has lower index than 1.21.3.

Steps to reproduce:

```
$ hetzner-k3s --version
1.0.7
```
```
cat k3s.yaml | grep -v \#
---
hetzner_token: 
cluster_name: test
kubeconfig_path: "./kubeconfig"
k3s_version: v1.21.3+k3s1
public_ssh_key_path: "~/.ssh/id_rsa.pub"
private_ssh_key_path: "~/.ssh/id_rsa"
use_ssh_agent: false
ssh_allowed_networks:
  - 0.0.0.0/0
api_allowed_networks:
  - 0.0.0.0/0
schedule_workloads_on_masters: false
masters_pool:
  instance_type: cpx21
  instance_count: 1
  location: nbg1
worker_node_pools:
  - name: small
    instance_type: cpx21
    instance_count: 3
    location: nbg1
additional_packages:
  - apparmor-profiles-extra
  - apparmor-profiles
  - apparmor-utils
  - apparmor
enable_encryption: true
```

```
$ hetzner-k3s upgrade -c k3s.yaml  --new-k3s-version v1.21.14+k3s1
Validating configuration...
Some information in the configuration file requires your attention:
  - New k3s version must be more recent than current version
```

**Comment by vitobotta** [comment]
Cool :)

**Comment by vitobotta** [comment]
@dzadonskii This is fixed in v1.0.8. It was a silly bug...
Closing for now but feel free to reopen if needed.
